### PR TITLE
core(report-generator): add url to CSV output

### LINKED ISSUE #10654: The CLI --output csv option should include the requestedUrl in the output.
[batch.txt](https://github.com/GoogleChrome/lighthouse/files/4547869/batch.txt)

[reports.zip](https://github.com/GoogleChrome/lighthouse/files/4547874/reports.zip)

I frequently batch up a number of pages using the CLI, using a command such as:

`lighthouse $page --quiet --output html --output csv --output json --only-categories performance --chrome-flags="--headless"`

This will produce a results with filenames such as www.example.com_2020-04-27_18-34-16.report,csv .json and .html

The reports all just have the domain as the filename. Within the html and json files, I can see what the page I tested was. However with the CSV file the total fields outputted for performance are related to the weightings and don't include the page tested.

There you can see from the attached that the page that was tested is not listed in the CSV, and as the filename doesn't include the page - only the domain - I have to jump through further hoops to compile the data together.

The json output includes, (like the HTML), the actual values rather than a weighting. If all of that could be included in the csv output it would be wonderful - but the page tested fields ("requestedUrl" and "finalUrl") would make the csv much easier to work with at a stroke.

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Pushing back against CSV as a thing is more work than making it better :)

**Related Issues/PRs**
closes #10654

### CODE REVIEW

File: lighthouse-core/report/report-generator.js [hunk 1]
```diff
@@ -68,13 +68,14 @@ class ReportGenerator {
...
-    const header = ['category', 'name', 'title', 'type', 'score'];
+    const header = ['requestedUrl', 'finalUrl', 'category', 'name', 'title', 'type', 'score'];
     const table = Object.values(lhr.categories).map(category => {
       return category.auditRefs.map(auditRef => {
         const audit = lhr.audits[auditRef.id];
         // CSV validator wants all scores to be numeric, use -1 for now
         const numericScore = audit.score === null ? -1 : audit.score;
-        return [category.title, audit.id, audit.title, audit.scoreDisplayMode, numericScore]
+        return [lhr.requestedUrl, lhr.finalUrl, category.title, audit.id, audit.title,
+            audit.scoreDisplayMode, numericScore]
```
Comment (connorjclark) [inline lighthouse-core/report/report-generator.js:78]: we gotta remove the push action. it doubles up on the lint errors

File: lighthouse-core/report/report-generator.js [hunk 2]
```diff
@@ -68,13 +68,14 @@ class ReportGenerator {
...
     // Possible TODO: tightly couple headers and row values
-    const header = ['category', 'name', 'title', 'type', 'score'];
+    const header = ['requestedUrl', 'finalUrl', 'category', 'name', 'title', 'type', 'score'];
     const table = Object.values(lhr.categories).map(category => {
       return category.auditRefs.map(auditRef => {
         const audit = lhr.audits[auditRef.id];
         // CSV validator wants all scores to be numeric, use -1 for now
         const numericScore = audit.score === null ? -1 : audit.score;
-        return [category.title, audit.id, audit.title, audit.scoreDisplayMode, numericScore]
+        return [lhr.requestedUrl, lhr.finalUrl, category.title, audit.id, audit.title,
```
Comment (brendankenny) [inline lighthouse-core/report/report-generator.js:77]: wait, so is there going to be two urls per line in the CSV?

File: lighthouse-core/report/report-generator.js [hunk 3]
```diff
@@ -68,13 +68,14 @@ class ReportGenerator {
...
+          audit.scoreDisplayMode, numericScore]
```
Comment (brendankenny) [inline lighthouse-core/report/report-generator.js:78]: > i respectfully disagree

yeah, I think that's an eslintrc bug

File: lighthouse-core/report/report-generator.js (hunk 2 above)
Comment (brendankenny) [reply lighthouse-core/report/report-generator.js:77]: > wait, so is there going to be two urls per line in the CSV?

I can live with that and don't care, just to be clear :)

Comment (patrickhulce) [reply lighthouse-core/report/report-generator.js:77]: correct there will be :)

**Pull request merged**